rts/lra/coordinator/src/test/java/io/narayana/lra/coordinator/tools/osb/mbean/ObjStoreBrowserLRATest.java

```diff
@@ -0,0 +1,93 @@
+import io.narayana.lra.coordinator.internal.LRARecoveryModule;
+import org.eclipse.microprofile.lra.annotation.LRAStatus;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.net.URI;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class ObjStoreBrowserLRATest {
+    private RecoveryManagerImple recoveryManager;
+
+    @Before
+    public void setUp() {
+        recoveryPropertyManager.getRecoveryEnvironmentBean().setRecoveryBackoffPeriod(1);
+        Implementations.install();
+        recoveryManager = new RecoveryManagerImple(false);
+        recoveryManager.addModule(new LRARecoveryModule());
+    }
+
+    @After
+    public void tearDown () {
+        recoveryManager.removeAllModules(false);
+        recoveryManager.stop(false);
+        Implementations.uninstall();
+    }
+
+    @Test
+    public void lraMBean() throws Exception {
+        ObjStoreBrowser osb = new ObjStoreBrowser();
+        LongRunningAction lra = new LongRunningAction(new Uid());
+
+        // creating the LRA records in the log store
+        lra.begin(Long.MAX_VALUE);
+        String coordinatorUrl = "http://localhost:8080/lra-coordinator";
+        String participantUrl = "http://localhost:8080/lra-participant";
+        LRAParticipantRecord lraParticipant = lra.enlistParticipant(URI.create(coordinatorUrl), participantUrl,
+                "/recover", Long.MAX_VALUE, null);
+        // initiating the ObjStoreBrowser
+        osb.start();
+
+        OSEntryBean jmxMBean = null;
+        try {
+            osb.probe();
+
+            UidWrapper uidWrapper = osb.findUid(lra.get_uid());
+            assertNotNull("Expected the LRA MBean uid was probed", uidWrapper);
+            jmxMBean = uidWrapper.getMBean();
+            assertNotNull("Expecting the UID contains the LRA jmx mbean", jmxMBean);
+            assertTrue("The provided jmx mbean should wrap ActionBean", jmxMBean instanceof ActionBean);
+            ActionBean actionBean = (ActionBean) jmxMBean;
+            assertEquals("One participant was enlisted", 1, actionBean.getParticipants().size());
+            LogRecordWrapper logRecord = actionBean.getParticipants().iterator().next();
+            assertTrue("The log wrapper needs to be from LRA", logRecord instanceof LRAParticipantRecordWrapper);
+            LRAParticipantRecordWrapper lraRecord = (LRAParticipantRecordWrapper) logRecord;
+            Assert.assertEquals("Participant should be active", LRAStatus.Active.name(), lraRecord.getStatus());
+            Assert.assertEquals("Compensator URI is expected as it was registered with '/compensate' suffix",
+                    participantUrl + "/compensate", lraRecord.getCompensator());
+        } finally {
+            if (jmxMBean != null) {
+                jmxMBean.remove(true);
```

Comment:
This works but why didn't you finish the LRA, and then re-probe, and then check that the beans for the LRA and participant are no longer present.